BTCPayServer/Payments/PayJoin/PayJoinRepository.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NBitcoin;
+using NBXplorer.Models;
+
+namespace BTCPayServer.Payments.PayJoin
+{
+    public class PayJoinRepository
+    {
+        HashSet<OutPoint> _Outpoints = new HashSet<OutPoint>();
```

Comment:
Don't you need a cryptocode key for this data to be segregated properly since the `SupportPayjoin` is on the network level?

Reply:
Maybe a WalletId is better. You also need to handle when a store changes their wallet, to clean the state for ongoing txs. This destroys data though